pallets/xcm-executor-utils/src/lib.rs

```diff
@@ -0,0 +1,252 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # XCM Executor Utils Pallet
+//!
+//! This is a utility pallet to help set the runtime parameters of XcmExecutor.
+//! Currently it offers an intuitive, on-chain maanger to set trust policies on
+//! incoming assets though `IsReserveFilter` and `IsTeleporterFilter`.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+pub mod weights;
+
+#[cfg(any(test, feature = "runtime-benchmarks"))]
+mod benchmarks;
+
+pub mod filters;
+
+pub use pallet::*;
+
+use {
+    crate::weights::WeightInfo,
+    frame_support::{pallet_prelude::*, DefaultNoBound},
+    frame_system::pallet_prelude::*,
+    serde::{Deserialize, Serialize},
+    staging_xcm::latest::{AssetId, MultiLocation},
+};
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use sp_runtime::BoundedVec;
+    use sp_std::vec::Vec;
+
+    // Default trust policies for incoming assets
+    #[derive(
+        PartialEq,
+        Eq,
+        Clone,
+        Encode,
+        Decode,
+        RuntimeDebug,
+        TypeInfo,
+        MaxEncodedLen,
+        Deserialize,
+        Serialize,
+    )]
+    pub enum DefaultTrustPolicy {
+        // Allow all incoming assets
+        All,
+        // Only allow assets native of the origin
+        AllNative,
+        // Do not allow any assets
+        Never,
+    }
+
+    #[derive(
+        DebugNoBound,
+        CloneNoBound,
+        EqNoBound,
+        PartialEqNoBound,
+        Encode,
+        Decode,
+        TypeInfo,
+        MaxEncodedLen,
+        Deserialize,
+        Serialize,
```

Comment:
This will not compile under no-std because `AssetId` only implements those traits if the std feature is enabled. Have you tested this branch in tanssi?

Reply:
Yes. I was about to ping you about it. Seems that Parity is allowing serialize/deserialize in no-std https://github.com/paritytech/polkadot/pull/7312 for https://github.com/paritytech/polkadot-sdk/issues/25.
@girazoki already had move serialize/deserialize out of the std gate for [upgrade 1.1.0](https://github.com/moondance-labs/tanssi/pull/272/files#diff-ee0709563521079ac120919640117a439cefd3429dba4a9e972f36af15cb7946L102)

Talked with him about it, and we decided to propose a PR upstream to serialize `AssetId` in no-std (`MultiLocation` [has been changed already](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/xcm/src/v3/multilocation.rs#L68), but weirdly `AssetId` hasn't)

Reply:
Open polkadot-sdk PR https://github.com/paritytech/polkadot-sdk/pull/3057